src/modules/EventModule.cpp

```diff
@@ -0,0 +1,792 @@
+							config.eventLowRiskLowRewardRunning = false;
+							config.eventTimer->stop();
+							UmikoBot::Instance().createMessage(message.channelId(), embed);
+						}
+						if (config.eventRaffleDrawRunning)
+						{
+							Embed embed;
+							embed.setColor(15844367);
+							embed.setTitle(eventEmoji);
+
+							if (config.currentTicketIndex == 0)
+							{
+								embed.setDescription("**RAFFLE DRAW** event has **ended**!\n"
+													"**There was no draw because no one bought tickets**");
+								config.claimedReward = true;
+								config.isEventRunning = false;
+								config.eventRaffleDrawRunning = false;
+								config.eventTimer->stop();
+								UmikoBot::Instance().createMessage(message.channelId(), embed);
+								return;
+							}
+
+							//Announce Winner
+							int lastTicket = static_cast<int>(config.currentTicketIndex);
+							std::random_device randomDevice;
+							std::mt19937 gen(randomDevice());
+							std::uniform_int_distribution<> dist(1, lastTicket);
+							int luckyTicketId = dist(gen);
+
+							QString tiketId = QString::number(luckyTicketId);
+							embed.setDescription("**RAFFLE DRAW** event has **ended**!\n"
+												"**The lucky ticket which wins the draw is `" + tiketId + "`**\n"
+												"**The owner can claim their rewards within 24 hours**");
+							config.claimedReward = false;
+							config.currentTicketIndex = 0;
+
+							//Get the user who has the lucky ticket
+							for (auto& user : raffleDrawGuildList[channel.guildId()])
+							{
+								for (auto& userTicket : user.m_TicketIds)
+								{
+									if (luckyTicketId == userTicket)
+									{
+										config.luckyUser = user.m_UserId;
+									}
+								}
+							}
+
+							//Clear all the tickets
+							for (auto& user : raffleDrawGuildList[channel.guildId()])
+								user.m_TicketIds.clear();
+
+							//Set the reward claiming time (24 hours)
+							config.raffleDrawRewardClaimTimer = new QTimer;
+							config.raffleDrawRewardClaimTimer->setInterval(24 * 3600000);
+							config.raffleDrawRewardClaimTimer->setSingleShot(true);
+
+							snowflake_t guildID = channel.guildId();
+							snowflake_t chan = channel.id();
+							snowflake_t authorId = message.author().id();
+							QObject::connect(config.raffleDrawRewardClaimTimer, &QTimer::timeout, [this, guildID, chan, authorId]()
+							{
+								auto& config = getServerEventData(guildID);
+								config.claimedReward = true;
+								UmikoBot::Instance().createMessage(chan, "The owner of the price in the RaffleDraw didn't get their rewards!\n"
+																		"The role(s) who have permission can start this event again!");
+							});
+							config.raffleDrawRewardClaimTimer->start();
+							config.isEventRunning = false;
+							config.eventRaffleDrawRunning = false;
+							config.eventTimer->stop(); // Forcefully stop the timer.
+							UmikoBot::Instance().createMessage(message.channelId(), embed);
+						}
```

Comment:
There is a lot of code duplication in end-event. You could instead manually trigger the timer timeout and handle ending in the timeout callback.

Reply:
Sorry, I didn't get your words.

Reply:
EDIT: I understand the code duplication thing, but I didn't understand what you want me to do.